osu.Game/Online/Chat/MessageFormatter.cs

```diff
@@ -207,38 +220,39 @@ public static LinkDetails GetLinkDetails(string url)
 
                     switch (args[1])
                     {
-                        case "chan":
+                        case @"chan":
                             linkType = LinkAction.OpenChannel;
                             break;
 
-                        case "edit":
+                        case @"edit":
                             linkType = LinkAction.OpenEditorTimestamp;
                             break;
 
-                        case "b":
+                        case @"b":
                             linkType = LinkAction.OpenBeatmap;
                             break;
 
-                        case "s":
-                        case "dl":
+                        case @"s":
+                        case @"dl":
                             linkType = LinkAction.OpenBeatmapSet;
                             break;
 
-                        case "spectate":
+                        case @"spectate":
                             linkType = LinkAction.Spectate;
                             break;
 
-                        case "u":
+                        case @"u":
                             return getUserLink(args[2]);
 
+                        case @"room":
+                            linkType = LinkAction.JoinRoom;
+                            break;
+
                         default:
                             return new LinkDetails(LinkAction.External, url);
                     }
 
                     return new LinkDetails(linkType, HttpUtility.UrlDecode(args[2]));
-
-                case "osump":
-                    return new LinkDetails(LinkAction.JoinMultiplayerMatch, args[1]);
```

Comment:
I removed this to avoid conflicts with stable and because @peppy stated at one point [that `osump` should not really be used going forward](https://discord.com/channels/90072389919997952/1083320217259749376/1365236568859611166).

If this sticks [we may want to consider pruning all mentions of `osump` from lazer code](https://github.com/search?q=repo%3Appy%2Fosu%20osump&type=code) and just leave well alone.

Reply:
I'd say that's a good direction, since we haven't really used it yet in a meaningful way (right?). Probably in a separate PR.